openmc/model/surface_composite.py

```diff
@@ -52,6 +53,168 @@ def __neg__(self):
+    outer_cyl : openmc.ZCylinder, openmc.YCylinder, or openmc.XCylinder
+        Outer cylinder surface.
+    inner_cyl : openmc.ZCylinder, openmc.YCylinder, or openmc.XCylinder
+        Inner cylinder surface.
+    plane1 : openmc.Plane
+        Plane at angle :math:`\\theta_1` relative to the first basis axis.
+    plane2 : openmc.Plane
+        Plane at angle :math:`\\theta_2` relative to the first basis axis.
+
+    """
+
+    _surface_names = ('outer_cyl', 'inner_cyl', 'plane1', 'plane2')
+
+    def __init__(self,
+                 r1,
+                 r2,
+                 theta1,
+                 theta2,
+                 center=(0.,0.),
+                 axis='z',
+                 **kwargs):
+
+        if r2 <= r1 * sqrt(2):
+            raise ValueError(f'r2 must be greater than r1.')
+
+        if theta2 <= theta1:
+            raise ValueError(f'theta2 must be greater than theta1.')
+
+        self._theta1 = theta1
+        self._theta2 = theta2
+
+        phi1 = pi / 180 * theta1
+        phi2 = pi / 180 * theta2
+
+        # Coords for axis-perpendicular planes
+        p1 = np.array([0., 0., 1.])
+
+        p2_plane1 = np.array([r1 * cos(phi1), r1 * sin(phi1), 0.])
+        p3_plane1 = np.array([r2 * cos(phi1), r2 * sin(phi1), 0.])
+
+        p2_plane2 = np.array([r1 * cos(phi2), r1 * sin(phi2), 0.])
+        p3_plane2 = np.array([r2 * cos(phi2), r2 * sin(phi2), 0.])
+
+        points = [p1, p2_plane1, p3_plane1, p2_plane2, p3_plane2]
+        if axis == 'z':
+            coord_map = [0, 1, 2]
+            self.inner_cyl = openmc.ZCylinder(*center, r1, **kwargs)
+            self.outer_cyl = openmc.ZCylinder(*center, r2, **kwargs)
+        elif axis == 'y':
+            coord_map = [1, 2, 0]
+            self.inner_cyl = openmc.YCylinder(*center, r1, **kwargs)
+            self.outer_cyl = openmc.YCylinder(*center, r2, **kwargs)
+        elif axis == 'x':
+            coord_map = [2, 0, 1]
+            self.inner_cyl = openmc.XCylinder(*center, r1, **kwargs)
+            self.outer_cyl = openmc.XCylinder(*center, r2, **kwargs)
+
+        for p in points:
+            p[:] = p[coord_map]
+
+        self.plane1 = openmc.Plane.from_points(p1, p2_plane1, p3_plane1,
+                                               **kwargs)
+        self.plane2 = openmc.Plane.from_points(p1, p2_plane2, p3_plane2,
+                                               **kwargs)
+
+    @classmethod
+    def from_theta_alpha(cls,
+                         r1,
+                         r2,
+                         theta,
+                         alpha,
+                         center = (0.,0.),
+                         axis='z',
+                         **kwargs):
+        """Alternate constructor for :attr:`CylinderSector`. Returns an
+        :attr:`CylinderSector` object based on a central angle :math:`\\theta`
+        and an angular offset :math:`\\alpha`. Note that
+        :math:`\\theta_1 = \\alpha` and :math:`\\theta_2 = \\alpha + \\theta`.
```

Comment:
raw-strings (`r"..."`) are your friend when you have lots of slashes
```suggestion
        r"""Alternate constructor for :class:`CylinderSector`. Returns a
        :class:`CylinderSector` object based on a central angle :math:`\theta`
        and an angular offset :math:`\alpha`. Note that
        :math:`\theta_1 = \alpha` and :math:`\theta_2 = \alpha + \theta`.
```

Reply:
Ooh, I didn't know about this! Good suggestion!